### PR TITLE
Support Docker deployments additionally

### PR DESCRIPTION
**Pull request by jcgruenhage**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ The following should be present on the target system
 | matrix_synapse_log_days_keep | 30 |
 | matrix_synapse_skip_tls | false |
 | matrix_synapse_registration_secret | _randomly generated_ |
+| matrix_synapse_deployment_method | pip | Either pip or docker [^1] |
+| matrix_synapse_supervision_method | systemd | Either systemd, runit or docker [^1] |
+
+[^1]: Docker must be used for both or neither deployment and supervision
```
Comment (madonius) [inline README.md]: Please use a proper reference rendered by Github, as this is the primary mean people get their information

File: handlers/main.yml [hunk 2]
```diff
@@ -2,15 +2,18 @@
 - name: "reload systemd"
   systemd:
     daemon_reload: yes
+  when: matrix_synapse_supervision_method == "systemd"
```
Comment (madonius) [inline handlers/main.yml]: This would be best handled in a block. It simply runs faster.

**Comment by madonius** [comment]
It would be great if you could provide a possibility to test the docker changes in Vagrant. I don't have that much time to explain it in detail atm. We could tackle that in person, if you like

File: README.md (hunk 1 above)
Comment (jcgruenhage) [reply README.md]: I thought that would be rendered by github, will fix it

**Comment by jcgruenhage** [comment]
I did expand the Vagrant test to include a playbook+machine for both pip and docker, so that both are tested

**Comment by jcgruenhage** [comment]
(And I did also test it both, still works)

**Comment by jcgruenhage** [comment]
Note to self: need to take care of handlers, the container is currently not restarted after config changes

**Pull request merged**